rules/cross-platform/discovery_kernel_module_enumeration.toml

```diff
@@ -0,0 +1,47 @@
+[metadata]
+creation_date = "2020/04/23"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/08/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+Loadable Kernel Modules (or LKMs) and Kernel Extensions are pieces of code that can be loaded and unloaded into the
+kernel upon demand. They extend the functionality of the kernel without the need to reboot the system. This identifies
+attempts to enumerate information about a kernel module or extension.
+"""
+false_positives = [
+    """
+    Security tools and device drivers may run these programs in order to enumerate kernel modules or extensions. Use of
+    these programs by ordinary users is uncommon. These can be exempted by process name or username.
+    """,
+]
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Enumeration of Kernel Modules or Extensions"
+note = "For macOS systems, Auditbeat must be run with an account that has access to the binaries in the /usr/sbin, /sbin, /usr/bin, and /bin directories. For Linux systems, to avoid conflicts with auditd, you will need to stop and disable auditd with 'sudo service stop auditd && sudo systemctl disable auditd'."
+risk_score = 47
+rule_id = "2d8043ed-5bda-4caf-801c-c1feb7410504"
+severity = "medium"
+tags = ["Elastic", "Linux", "macOS"]
+type = "query"
+
+query = '''
+event.category:process and event.type:start and process.args:(((kmod and list) or (kmod and static-nodes)) or (depmod or lsmod or modinfo or kextstat))
```

Comment:
```suggestion
event.category:process and event.type:start and 
  process.args:(depmod or lsmod or modinfo or kextstat or kmod and (list or "static-nodes"))
```

I also see you removed all of the sudo. I know there were quarks that @bm11100 had noticed around it. If it is a matter of duplicity, would referencing the UID as 0 work (unless it was another issue altogether).

Reply:
just adding a note on systemextensionsctl - thought not exactly a kernel extension (so maybe for a separate future rule)

Reply:
That _may_ work for the endpoint as it added a `user.Ext.real.id` field, which we may be able to wrap some `not`s around it.

Agreed, this definitely had some "quarks" around it, to put it gently. 

**Auditbeat** (no dice)
![image](https://user-images.githubusercontent.com/7442091/92036246-8cd46200-ed35-11ea-93ca-045278967a07.png)

**Endpoint**
![image](https://user-images.githubusercontent.com/7442091/92036009-35ce8d00-ed35-11ea-80b5-3c727570b332.png)